service/history/workflowResetor.go

```diff
@@ -353,51 +397,42 @@ func (w *workflowResetor) setEventIDsWithHistory(msBuilder mutableState) int64 {
 	return firstEvent.GetEventId()
 }
 
-func (w *workflowResetor) generateReplicationTasksForReset(terminateCurr bool, currMutableState, newMutableState mutableState, domainEntry *cache.DomainCacheEntry) []persistence.Task {
+func (w *workflowResetorImpl) generateReplicationTasksForReset(terminateCurr bool, currMutableState, newMutableState mutableState, domainEntry *cache.DomainCacheEntry) []persistence.Task {
 	var repTasks []persistence.Task
 	if newMutableState.GetReplicationState() != nil {
 		if terminateCurr {
-			firstEventIDForNew := w.setEventIDsWithHistory(newMutableState)
+			// we will 2 replication tasks for this case
 			firstEventIDForCurr := w.setEventIDsWithHistory(currMutableState)
```

Comment:
we generate will 2 replication tasks for this case

Reply:
fixed